packages/react/src/settings/hooks/useConfiguration.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  fetchConfiguration as fetchConfigurationAction,
+  getConfiguration,
+  getConfigurationError,
+  isConfigurationFetched,
+  isConfigurationLoading,
+  StoreState,
+} from '@farfetch/blackout-redux';
+import { useCallback, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import useAction from '../../helpers/useAction';
+import type { Configuration } from '@farfetch/blackout-client';
+import type { UseConfigurationOptions } from './types';
+
+const useConfiguration = (
+  configurationCode: Configuration['code'],
+  options: UseConfigurationOptions = {},
+) => {
+  const { enableAutoFetch = true, fetchQuery, fetchConfig } = options;
+
+  const error = useSelector((state: StoreState) =>
+    getConfigurationError(state, configurationCode),
+  );
+  const isLoading = useSelector((state: StoreState) =>
+    isConfigurationLoading(state, configurationCode),
+  );
+  const isFetched = useSelector((state: StoreState) =>
+    isConfigurationFetched(state, configurationCode),
+  );
+  const configuration = useSelector((state: StoreState) =>
+    getConfiguration(state, configurationCode),
+  );
+  const fetch = useAction(fetchConfigurationAction);
+
+  const fetchConfiguration = useCallback(
+    () => fetch(configurationCode, fetchQuery, fetchConfig),
```

Comment:
I missed this: Can you add a check for `configurationCode` to make sure it is defined before executing fetch?
Something like this:

```suggestion
    () => {
      if(!configurationCode) {
        return Promise.reject(new Error('Invalid parameter `configurationCode` for `fetch`'));
       }
       
       return fetch(configurationCode, fetchQuery, fetchConfig);
    }
```

Add a corresponding test as well please when `useConfiguration` is called without a configurationCode.